### DISCUSSION TITLE
UI on mobile

### DISCUSSION THREAD

**Discussion by christophedc0** [obiwan007/despresso#69] [Beta]
As Joe has mentioned on discord, the UI (specifically landscape mode) on mobile doesn't scale well.

> The app was designed to be on a tablet, to replace the current chinese android tablet.

**Comment by obiwan007** [comment]
It will not be perfect on any phone type device. I just made it a bit better usable in the espresso screen.
But it is still tablet first for now.

**Reply by razmiy** [reply]
Would it be possible to change the resolution?

I have an old Huawei 7" tablet with the resolution of 1200 x 1920. Everything else is running well except for the displayed screen resolution. Looks rather bad like it was too big for the screen

**Reply by obiwan007** [reply]
The app is actually automatially scaling. There is no option to change resolutions.
It should scale according to your text settings in Android to make fonts more readable.
Could you post a screenshot here how it is looking?

In iOS you have setting for dynamic test size. This is taken into account in the app. Not sure how it is called in android atm.

**Reply by razmiy** [reply]
Some images. The app actually works but I just thought that it looks better on my ipad and xiaomi mi pad 2
![image](https://github.com/obiwan007/despresso/assets/69907817/35b7f003-749a-4073-b0b1-8c170f850688)

**Reply by razmiy** [reply]
On xiaomi mi pad 2
![image](https://github.com/obiwan007/despresso/assets/69907817/e3efd564-b2a2-4114-996c-ef095b7e6896)

**Reply by obiwan007** [reply]
Yes pls reduce the font scaling. Usually this is located in the accessibility settings. I have no android here at the moment to check this .

**Reply by razmiy** [reply]
Tried that. No go, the fonts maybe smaller but there is no change in the other aspects of the gui layout

**Reply by obiwan007** [reply]
The layout is fixed. The screen is split into three columns and the elements in the columns take the space they need.
The only thing changing is the font size. As text is the most space consuming thing on the recipe page this should make it a bit smaller and give room to other elements like the graph.
Bit maybe the DPI of the tablet is solowthat it looks not nice. Difficult to have an app running on all kind of devices with all kind of different resolutions and DPI settings...

**Reply by razmiy** [reply]
![image](https://github.com/obiwan007/despresso/assets/69907817/e5ceffb6-11ea-42a2-a3be-f381ec77c682)

**Reply by razmiy** [reply]
The resolution of this tablet is 1200 x 1920. So hardly low

**Reply by obiwan007** [reply]
Can you give me the exact type of the tablet? This resolution on a 7inch sounds astonishinly high.

**Reply by razmiy** [reply]
![image](https://github.com/obiwan007/despresso/assets/69907817/d3ee34bf-d1b4-41ac-8ec0-4105280376d8)

**Reply by razmiy** [reply]
Maybe it's not really the resolution issue at all but more on the aspect ratio of the display. 

The app appears to works better on screen with 4:3 aspects ratio like the ipad and the xiaomi mi pad and not the 16:10 ratio as my 7" tablet, or worst on any high resolution phone screen with 16:9 ratio.

From my observation - on ratio other than the 4:3, the top and bottom horizontal panels just does not scale well enough and thus hogging more of the screen real estate

**Reply by obiwan007** [reply]
Yes that might be. flutter tries to keep touch surfaces the same size regardless of the resolution. Otherwise a very high DPI system would have too small and not touchable elements like buttons. So every button on Android has a specific absolute size which is not changing if DPI changes.
Not sure if I would like to change that behaviour.
ButI found a lib which could help to scale stuff a bit easier and make some elements a bit more space restricted. Not sure how well this will work.
The app is developed with ipad in mind. And it is working well on 10inc tabs like my ipad air and Galaxy tab a8. Teclast is also not so perfect looking. We will see ugh ow this could be solved.

**Reply by razmiy** [reply]
Thanks and no worries, i have managed to make it a little bit presentable and I suppose it is already okay and useable for now on the 7". There are other things however that i'm having a bit of problem, like not being able to upload to visualizer, the disappearing history and the patchy integration with my mqtt/ha. Anyway, i'll keep on using it to understand it better. Thanks again for your efforts

**Reply by obiwan007** [reply]
Please feel free to Crete issues for this.
Visualizer is working perfect if you enter the correct credentials. Make sure upper lowercase for username is right as visualizer is case sensitive for some reason in usernames.

Which history is disappearing? If you uninstall the app the history is gone. You need to create backup to prevent this.

What is the problem with mqtt? Works fine with iobroker.

**Reply by razmiy** [reply]
I'm having the following error every time when trying unsuccessfully to upload to visualizer:/

Error uploading shot: HandshakeException: Handshake error in client (OS Error:
CERTIFICATE_VERIFY_FAILED: certificate has expired(handshake.cc:393))

What can be done to prevent this from happening?

**Reply by obiwan007** [reply]
Was it working before?
Check the exact spelling of your username and password. Your username might be in uppercase which wouöd prevent the uploads...
Just uploaded a shot with latest build. It should work. 
Maybe visualizer updated it's certificate in the moment you tested?

**Reply by razmiy** [reply]
It has never worked for me on this app although it works flawlessly in the native app, and yes my spelling for my credentials is fine

**Reply by obiwan007** [reply]
Which Android version? 
Zur de1app is probably using a http connection which is not using any certificates. 
We are using a secure channel so this might be the difference.
I see one event in our log to point to an Android 6 tablet. This is probably a bit to old....

**Comment by razmiy** [comment]
Yes, I think it’s the android thing and i am using android 6 at the moment. Those tablets that i was using are indeed ancient 😝. I’ll be doing some testing on a more recent android later

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________
